task-reminder-client/libs/states/user/src/lib/guard/auth-guard.service.ts

```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@angular/core';
+import { Router, CanActivate } from '@angular/router';
+import { Store } from '@ngrx/store';
+import { map, Observable, of, tap } from 'rxjs';
+import * as UserSelectors from '../+state/user.selectors';
+
+@Injectable({ providedIn: 'root' })
+export class AuthGuardService implements CanActivate {
+
+  token$ = this.store.select(UserSelectors.getToken);
+
+  constructor(
+        private router: Router,
+        private store : Store
+    ) { }
+
+    canActivate(): Observable<boolean> {
+      return this.token$.pipe(
+        map(token => token !== ''),
+        tap((token) => {
```

Comment:
after **map** operator, the param is no longer a token (string) value. It's boolean value instead, should use a better param name. Ex: hasToken